lib/_http_server.js

```diff
@@ -482,8 +483,15 @@ function connectionListener(socket) {
 
       res.detachSocket(socket);
 
-      if (res._last) {
-        socket.destroySoon();
+      if (res._last) socket.destroySoon();
+      else if (outgoing.length === 0) {
+        // When outgoing queue is empty we will wait for
+        // server.keepAliveTimeout and destroy socket not emiting
+        // 'timeout' events in req, res and server objects
+        socket.removeAllListeners('timeout');
+        socket.setTimeout(self.keepAliveTimeout, function onKeepAliveTimeout() {
```

Comment:
Technically this TCP connection is timing out, right? The other side it supposed to close it once it is done with it. If it does not - we kill it, and let API users know about it.

I think it makes sense.